server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/FilterAllocationDecider.java

```diff
@@ -125,52 +125,61 @@ public Decision canRemain(ShardRouting shardRouting, RoutingNode node, RoutingAl
 
         return allocation.decision(Decision.YES, NAME, "node passes include/exclude/require filters");
     }
 
     private Decision shouldFilter(IndexMetadata indexMd, DiscoveryNode node, RoutingAllocation allocation) {
-        Decision decision = shouldClusterFilter(node, allocation);
+        Decision decision = shouldIndexFilter(indexMd, node, allocation);
         if (decision != null) return decision;
 
-        decision = shouldIndexFilter(indexMd, node, allocation);
+        decision = shouldClusterFilter(node, allocation);
         if (decision != null) return decision;
 
         return allocation.decision(Decision.YES, NAME, "node passes include/exclude/require filters");
     }
 
     private Decision shouldIndexFilter(IndexMetadata indexMd, DiscoveryNode node, RoutingAllocation allocation) {
         if (indexMd.requireFilters() != null) {
-            if (indexMd.requireFilters().match(node) == false) {
+            if (indexMd.requireFilters().match(node)) {
+                return allocation.decision(Decision.YES, NAME, "node matches index setting [%s] filters [%s]",
+                    IndexMetadata.INDEX_ROUTING_REQUIRE_GROUP_PREFIX, indexMd.requireFilters());
+            } else {
                 return allocation.decision(Decision.NO, NAME, "node does not match index setting [%s] filters [%s]",
                     IndexMetadata.INDEX_ROUTING_REQUIRE_GROUP_PREFIX, indexMd.requireFilters());
             }
         }
         if (indexMd.includeFilters() != null) {
-            if (indexMd.includeFilters().match(node) == false) {
-                return allocation.decision(Decision.NO, NAME, "node does not match index setting [%s] filters [%s]",
+            if (indexMd.includeFilters().match(node)) {
+                return allocation.decision(Decision.YES, NAME, "node matches index setting [%s] filters [%s]",
                     IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_PREFIX, indexMd.includeFilters());
+            } else {
+                return allocation.decision(Decision.NO, NAME, "node does not match index setting [%s] filters [%s]",
+                    IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_PREFIX, indexMd.requireFilters());
             }
         }
         if (indexMd.excludeFilters() != null) {
-            if (indexMd.excludeFilters().match(node)) {
-                return allocation.decision(Decision.NO, NAME, "node matches index setting [%s] filters [%s]",
+            if (indexMd.excludeFilters().match(node) == false) {
+                return allocation.decision(Decision.YES, NAME, "node does not match index setting [%s] filters [%s]",
                     IndexMetadata.INDEX_ROUTING_EXCLUDE_GROUP_SETTING.getKey(), indexMd.excludeFilters());
+            } else {
+                return allocation.decision(Decision.NO, NAME, "node matches index setting [%s] filters [%s]",
+                    IndexMetadata.INDEX_ROUTING_EXCLUDE_GROUP_SETTING, indexMd.requireFilters());
```

Comment:
```suggestion
                    IndexMetadata.INDEX_ROUTING_EXCLUDE_GROUP_SETTING.getKey(), indexMd.excludeFilters());
```

Reply:
I wonder why this is outdated and marked as resolved while it hasn't changed (still missing `.getKey()`).
Another solution: use `INDEX_ROUTING_EXCLUDE_GROUP_PREFIX` instead like done at the other settings (would be even better for consistency and also would not print a `.` at the end).

Reply:
👍